### PR TITLE
Bump glimmer-vm packages to 0.24.

### PR DESCRIPTION
**Pull request by chancancode**
* [x] ESLint (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=ESLint)
* [x] Static Content (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=static%20content)
* [x] Dynamic Content (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=dynamic%20content)
* [x] Syntax (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=syntax)
* [x] Helpers (http://localhost:5200/tests/index.html?hidepassed&package=ember-glimmer&filter=helpers)
    * [x] Custom Helpers (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=custom%20helpers)
    * [x] {{if}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bif%7D%7D)
    * [x] {{unless}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bunless%7D%7D)
    * [x] {{input}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%3A%20%7B%7Binput)
    * [x] {{textarea}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Btextarea%7D%7D)
    * [x] {{partial}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bpartial%7D%7D)
    * [x] {{yield}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Byield%7D%7D)
    * [x] {{hash}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bhash%7D%7D)
    * [x] {{concat}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bconcat%7D%7D)
    * [x] {{log}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Blog%7D%7D)
    * [x] {{loc}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bloc%7D%7D)
    * [x] {{get}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bget%7D%7D)
    * [x] {{render}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Brender%7D%7D)
    * [x] {{mut}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Bmut%7D%7D)
    * [x] {{readonly}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=%7B%7Breadonly%7D%7D)
    * [x] closure {{action}} (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=closure%20%7B%7Baction%7D%7D)
    * [x] element action (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=element%20action)
* [x] Components (http://localhost:5200/tests/index.html?package=ember-glimmer&filter=Components%20test&hidepassed)
    * [x] this.attrs, positional arguments
        * We broke this in https://github.com/emberjs/ember.js/commit/d979a63f3fb9521a5ae6ae46fef017b8868ab580
    * [x] Cascading failures (cannot reuse the same DIV, cannot start nested transactions, etc)
    * [x] contextual components (http://localhost:7200/tests/index.html?hidepassed&filter=contextual)
    * [x] local lookup (http://localhost:7200/tests/index.html?hidepassed&filter=local)
* [x] Iterable (http://localhost:5200/tests/index.html?package=ember-glimmer&module=Iterable)
    * Remove? (https://github.com/emberjs/ember.js/pull/14763#issuecomment-301955781)
* [x] ...

### CODE REVIEW

**Comment by knownasilya** [comment]
Interesting seeing the dog fooding of component-managers and how they are used. Good stuff and much to learn.

File: packages/ember-glimmer/lib/component-managers/curly.js [hunk 1]
```diff
@@ -103,13 +96,65 @@ class CurlyComponentLayoutCompiler {
 
 CurlyComponentLayoutCompiler.id = 'curly';
 
+export class PositionalArgumentReference {
```
Comment (rwjblue) [inline packages/ember-glimmer/lib/component-managers/curly.js]: Need to conform to the `PathReference` interface (missing `get` at least ATM).

**Comment by rwjblue** [comment]
Did some quick asset size checks, and the `min + gz` size is slightly better (current `master` `120.54 KB`; after this PR `120.15 KB`).

**Comment by rwjblue** [comment]
I did some general benchmarking (against the emberaddons.com app), and it seems that we haven't introduced a regression here.  

phases

Seems good to ship...

**Comment by knownasilya** [comment]
:shipit:

**Comment by rwjblue** [comment]
Did another test against emberobserver.com (which has a larger rendering phase):

phases

File: packages/ember-glimmer/lib/component-managers/outlet.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+  didRenderLayout(bucket) {
+    bucket.finalize();
+
+    if (DEBUG) {
+      this.debugStack.pop();
+    }
+  }
+
+  didCreateElement() {}
```
Comment (chadhietala) [inline packages/ember-glimmer/lib/component-managers/outlet.js]: Can't these go on the abstract manager?

File: packages/ember-glimmer/lib/component-managers/render.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+  getTag() {
+    return null;
+  }
+
+  getDestructor() {
+    return null;
+  }
+
+  didCreateElement() {}
```
Comment (chadhietala) [inline packages/ember-glimmer/lib/component-managers/render.js]: Same comment about putting them on the abstract manager.

Comment (rwjblue) [inline packages/ember-glimmer/lib/component-managers/render.js]: Yep, totally agreed.

**Comment by chadhietala** [comment]
LGTM

**Comment by knownasilya** [comment]
Woop

**Pull request merged**